ding_dibs.email_reports.inc

```diff
@@ -0,0 +1,251 @@
+    ->execute()
+    ->fetchAll();
+  $max_count = count($report);
+  if (!$max_count) {
+    return '<p>' . t('No transactions found.') . '</p>';
+  }
+  for ($count = 0; $count < $max_count; $count++) {
+    //Clipping of output orders with the status of "1/1/1" and "0/0/0"
+    if (!($report[$count]->payment_status == $report[$count]->provider_payment_status && $report[$count]->payment_status == $report[$count]->dibs_capture_status && $report[$count]->provider_payment_status == $report[$count]->dibs_capture_status)) {
+      $report_valid[] = $report[$count];
+    }
+  }
+  $header = array(
+    array('data' => t('Order ID')),
+    array('data' => t('Transaction ID')),
+    array('data' => t('Date')),
+    array('data' => t('User')),
+    array('data' => t('Text')),
+    array('data' => t('Amount')),
+    array('data' => t('Cleared')),
+    array('data' => t('Paid')),
+    array('data' => t('Captured')),
+  );
+  $rows = array();
+  foreach ($report_valid as $record) {
+    $rows[] = _ding_dibs_report_generation_values($record);
+  }
+
+  $output = theme('table', array(
+    'header' => $header,
+    'rows' => $rows,
+    'attributes' => array('id' => 'ding_dibs_email_reports'),
+  ));
+  return $output;
+}
+
+/**
+ * Callback that builds page reports run cron.
+ *
+ * @return
+ *   Returns tabular representation of values results.
+ */
+function ding_dibs_cron_report() {
+  $report = db_select('ding_dibs_report_cron', 'd')
+    ->fields('d', array('id', 'time_start', 'status', 'text'))
+    ->orderBy('id', 'DESC')
+    ->execute()
+    ->fetchAll();
+  $max_count = count($report);
+  if (!$max_count) {
+    return drupal_set_message(t('No cron report.'), 'warning');
+  }
+  drupal_add_css(drupal_get_path('module', 'ding_dibs') . '/css/ding_dibs.css');
+  $header = array(array('data' => t('Message'), 'class' => 'payment'));
+  $rows = array();
+  foreach ($report as $record) {
+    if (empty($record->text)) {
+      $day = format_date($record->time_start, 'custom', 'd', date_default_timezone_get());
+      $hour = format_date($record->time_start, 'custom', 'H:i', date_default_timezone_get());
+      $month = format_date($record->time_start, 'custom', 'F', date_default_timezone_get());
+      $record->text = $day . 'th ' . $month . ' ' . $hour . ': Job failed.';
+    }
+    $row = array(
+      array(
+        'data' => t($record->text),
+        'class' => 'payment',
+      ),
+    );
+    $rows[] = $row;
+  }
+  $output = theme('table', array(
+    'header' => $header,
+    'rows' => $rows,
+  ));
+  return $output;
+}
+
+/**
+ * The function adds to the dataset for the numbers zero to ten.
+ *
+ * @param $item
+ *   An array of input data for the time period.
+ *
+ * @return
+ *   Returns an array with the format of the time 01, 02.
+ */
+function _ding_dibs_generate_zero($item) {
+  return str_pad($item, 2, "0", STR_PAD_LEFT);
+}
+
+/**
+ * The function generates a table with data reporting database table
+ * so dibs transactions.
```

Comment:
I appreciate the effort to add some documentation here but it is not very helpful.

1. The DocBlock is invalid. The function only accepts a single parameter.
2. The description of each field on the transaction object does not add anything that is not already described in [`ding_dibs_schema`](http://cgit.drupalcode.org/dibs/tree/dibs.install#n36)
3. The wording has several problems:
  - "The function generates a table" - it generates a table row
 -  "with data reporting database table so dibs transactions." - to be honest this makes no sense